service/src/main/java/bio/terra/workspace/service/workspace/flight/WorkspaceDeleteFlight.java

```diff
@@ -73,35 +71,53 @@ public WorkspaceDeleteFlight(FlightMap inputParameters, Object applicationContex
         new DeleteGcpProjectStep(
             appContext.getCrlService(), appContext.getGcpCloudContextService()),
         cloudRetryRule);
+
+    addStep(
+        new EnsureNoWorkspaceChildrenStep(appContext.getSamService(), userRequest, workspaceUuid));
```

Comment:
I was originally thinking that we would check if any Sam child resources existed that we would not be able to delete _before_ we called `DeleteControlledAzureResourcesStep` and `DeleteControlledSamResourcesStep` (so that the workspace would stay in its original state). But is there no way for us to know what is deletable without actually first attempting to delete things?

Reply:
I don't see any way to know, especially as we might be depending upon external services to actually delete resources, and some of those might have child resources, etc.